### PR TITLE
[Mixpanel] DUOS-2521 User Registration and Sign In

### PR DESCRIPTION
**Pull request by gabelmattos**
### Addresses
Mixpanel implementation

### Summary
The latest updates to DUOS-UI includes the addition of Bard's Methods for Mixpanel Calls and Registration and Sign-in events. There was also some minor refactoring done on the SingIn.js file to improve readability.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by gabelmattos** [comment]
The `SignIn.js` file could use further refactoring, we need to do some separation of concerns and put some of the functions in a utility file. I recommend we do it in a separate ticket and PR.

File: src/libs/config.js [hunk 1]
```diff
@@ -7,6 +7,8 @@ export const Config = {
 
   getApiUrl: async () => (await getConfig()).apiUrl,
 
+  getBardApiUrl: async () => (await getConfig()).bardApiUrl, // Mixpanel
+
```
Comment (rushtong) [inline src/libs/config.js:11]: We should have the terra-helmfile PR in place before this PR is finally merged.

File: src/libs/metrics.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  const config = {
+    method: 'POST',
+    url: `${await getBardApiUrl()}/api/identify`,
+    data: body,
+    headers: { Authorization: `Bearer ${Storage.getGoogleData()?.accessToken}` },
+    signal,
+  };
+
+  return axios(config).catch(() => { });
```
Comment (rushtong) [inline src/libs/metrics.js:87]: We actually could use a new error reporting feature, namely, report to Sentry. That's very much beyond the scope of this PR, but perhaps we could stub that out here for future implementation. We do have a `StackdriverReporter` but that should be deprecated in favor of a Sentry-based version.

**Review by rushtong — APPROVED** [review]
Overall, looks good! Some minor comments inline and we still need a terra-helmfile PR before this gets merged. 👍🏽

File: src/libs/events.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const eventList = {
+  userRegister: 'user:register',
+  userSignIn: 'user:signin',
+
+  pageView: 'page:view',
```
Comment (connorlbark) [inline src/libs/events.js:5]: Are the pageView hooks to be implemented in a future ticket?

**Review by connorlbark — APPROVED** [review]
Looks great! Sign in page has some great cleanup. The requests were all sent properly locally when I ran it.

**Comment by rushtong** [comment]
I missed a couple things in my earlier review. We need similar config changes to:
* config/alpha.json
* config/base-config.json

File: src/libs/events.js (hunk 3 above)
Comment (gabelmattos) [reply src/libs/events.js:5]: Yeah, pageviews should be included in future tickets.

**Pull request merged**